### ISSUE TITLE
Recaptcha Not Rendering... Out of ideas

### ISSUE THREAD

**Issue by bfjare** [solspace/craft-freeform#337]
Hey Kelsey,

I have recaptcha installed and its not rendering and I cant figure out what I'm missing and was hoping you might have some ideas on debugging this.

Here is what I have done.

1. Triple checked that I was using V2 checkbox on both ends in craft and recaptcha
2. Installed the demo templates and still have the issue there
3. Created a test template with only the code below
4. Check the console and there are no javascript errors

`{% set form = craft.freeform.form("secureMessage") %}

{% if form.submittedSuccessfully %}
  <div>You've successfully submitted this form!</div>
{% endif %}

{{ form.render }}`

I cant think of anything else, do you have any ideas?

**Comment by kjmartens** [comment]
Hi @bfjare,

Sorry for the trouble you're experiencing right now.

Thanks for letting me know what you've tried again. Have you dragged and dropped the special reCAPTCHA field into your form layout in the form builder? This is one additional step that is necessary for _reCAPTCHA v2 Checkbox_, since you get to control where it is positioned. 🙂

**Comment by bfjare** [comment]
Yep I did do that its creating the html container for the checkbox to be loaded. I can see it using the inspect tool in Chrome, but the actual checkbox is not rendering.

**Comment by kjmartens** [comment]
Ok thanks. 🙂 

Have you also added your site domain to your account on the reCAPTCHA site?

Also, inside the Network tab of your browser, are there any items that show as red? And if so, what does the Response sub-tab show for it?

**Comment by bfjare** [comment]
Yep thats been done, and no errors in the network tab

**Comment by kjmartens** [comment]
Can you confirm which exact version of Craft and Freeform you're using?

**Comment by bfjare** [comment]
Craft CMS 3.7.40.1
Freeform 3.13.9

**Comment by Zw2b4ck** [comment]
We face the exact same issue. The recaptcha is working locally, but not on the deployed version.
It creates the container for the captcha, but doesnt render it.

Recaptcha V2 standard (visible)
Is added to form.


Freeform 3.13.9
Craft CMS 3.7.42

**Comment by kjmartens** [comment]
@Zw2b4ck Since Flatpickr also isn't working, I wonder if you're experiencing the issue where the server rules you have applied are affecting Freeform's JS and CSS files. The default for Freeform is to load its scripts as static URLs, which becomes a problem in this case. Does switching the "Freeform Script Insert Type" setting (Craft CP -> Freeform -> Settings -> Form Behavior) to "As Files" make any difference? 🙂 

@bfjare Probably a long shot, but are you able to try switching this setting as well and see if it makes any difference at all?

**Comment by bfjare** [comment]
I just gave this a shot and it didnt work.  I do however think it could be a file permissions issue but not sure where to look considering I'm not getting errors/

**Comment by bfjare** [comment]
Ok thats interesting.  I do not have any errors in the network tab.  However I am also not seeing the api.js file either, no matter what form behavior setting  I choose.

**Comment by kjmartens** [comment]
@bfjare If you place this code manually in the `<head>`, does reCAPTCHA work for you?

```
<script src="https://www.google.com/recaptcha/api.js" async defer></script>
```

**Comment by kjmartens** [comment]
@bfjare Odd that this is necessary for it to work for you. Are you able to send a screenshot of what your Diagnostics page looks like in the Freeform settings? And out of curiosity, do other JS-related features in Freeform work (you can test this by adding something like a Date field with picker, Conditional Rules, etc)? I'm wondering if all Freeform JS is not working, or just reCAPTCHA.

**Comment by Zw2b4ck** [comment]
> @Zw2b4ck Since Flatpickr also isn't working, I wonder if you're experiencing the issue where the server rules you have applied are affecting Freeform's JS and CSS files. The default for Freeform is to load its scripts as static URLs, which becomes a problem in this case. Does switching the "Freeform Script Insert Type" setting (Craft CP -> Freeform -> Settings -> Form Behavior) to "As Files" make any difference? 🙂
> 
> @bfjare Probably a long shot, but are you able to try switching this setting as well and see if it makes any difference at all?

--> switching the "Freeform Script Insert Type" setting (Craft CP -> Freeform -> Settings -> Form Behavior) to "As Files". WORKED ! It solved the issue with the captcha, same as the datepicking issue! Im happy! :)

Thanks a lot for the fast help!

**Comment by kjmartens** [comment]
@Zw2b4ck Happy to help! Glad it's fixed now. 🙂
